backend/src/tests.py

```diff
@@ -148,3 +150,26 @@ def test_rrule_validation_failure(self):
         }
         response = self.client.post("http://testserver/api/rules", params={"userid": "testuser"}, json=body)
         self.assertEqual(response.status_code, 400)
+
+    def test_future_start_end_dates(self):
+        response = self.client.post("http://testserver/api/transactions", params={"userid": "testuser", "currentBalance": "0", "startDate": "2020-1-1", "endDate": "2021-1-1"})
+        self.assertEqual(response.status_code, 400)     
+
+    def test_start_date_preceeds_end_date(self):
+        now = datetime.now().date()
+        endDate = now - relativedelta(days=1)      
+        response = self.client.post("http://testserver/api/transactions", params={"userid": "testuser", "currentBalance": "0", "startDate": now.strftime("%Y-%m-%d"), "endDate": endDate.strftime("%Y-%m-%d")})
+        self.assertEqual(response.status_code, 400)
+
+    def test_future_limit_start_date(self):
+        now = datetime.now().date()
+        startDate = now + relativedelta(years=3, days=1)
+        endDate = now + relativedelta(years=3, days=2)        
+        response = self.client.post("http://testserver/api/transactions", params={"userid": "testuser", "currentBalance": "0", "startDate": startDate.strftime("%Y-%m-%d"), "endDate": endDate.strftime("%Y-%m-%d")})
+        self.assertEqual(response.status_code, 400)
+
+    def test_future_limit_end_date(self):
```

Comment:
Need to test missing query parameters and no query parameters.

I believe the validations you are doing are done in handler.py. Could we refactor to take out the stuff in handler.py? Calling handler was a hack on my part.

Reply:
(The handler refactor is not a required ask, but query params is required for the story. You decide whether it's this PR or not.)

Reply:
Testing missing and no query parameters has been addressed in the latest push.